File: app/src/main/res/LoginActivity/app/src/main/java/com/example/loginactivity/MainActivity.java
package com.example.loginactivity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.content.Intent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;


public class MainActivity extends AppCompatActivity {

    private EditText edUsername;
    private Button btnLogin;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        edUsername=findViewById(R.id.ed_username);
        btnLogin=findViewById(R.id.btn_login);
        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String strUsername=edUsername.getText().toString();
                Intent intent=new Intent(MainActivity.this, WelcomeActivity.class);
                intent.putExtra("Username",strUsername);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun createBirthdayCard(view: View) {
       val name= findViewById<EditText>(R.id.userName).editableText.toString()
        val intent= Intent(this, BirthdayGreetingsActivity::class.java)
        intent.putExtra(BirthdayGreetingsActivity.NAME_EXTRA, name)
        startActivity(intent)
    }


}